index.d.ts

```diff
@@ -0,0 +1,758 @@
+declare module 'extra-array' {
+  /**
+   * Binary searches leftmost value in sorted array.
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} first index of value | ~(index of closest value)
+   */
+  export function bsearch<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Binary searches value in sorted array.
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} index of value | ~(index of closest value)
+   */
+  export function bsearchAny<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Binary searches closest value in sorted array.
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} index of closest value
+   */
+  export function bsearchClosest<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Binary searches rightmost value in sorted array.
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} last index of value | ~(index of closest value)
+   */
+  export function bsearchRight<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Breaks array into chunks of given size.
+   * @param {Array} x an array
+   * @param {number?} n chunk size
+   * @returns {Array<Array>} chunks
+   */
+  export function chunk<T>(x: T[], n?: number): T[][];
+
+  /**
+   * Compares two arrays.
+   * @param {Array} x an array
+   * @param {Array} y another array
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} x<y: -1, x=y: 0, x>y: 1
+   */
+  export function compare<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Appends arrays to the end.
+   * @param {Array} x an array (updated)
+   * @param {...Iterable} ys arrays to append
+   * @returns {Array} x
+   */
+  export function concat$<T>(x: T[], ...ys: Iterable<T[]>[]): T[];
```

Comment:
Does not mean a list of iterables?
`...ys: Iterable<T[]>[]`

Each y can be an array or iterable, like:

```javascript
var x = [1, 2];
array.concat$(x, [3,  4]);
```

I think it should be

```suggestion
  export function concat$<T>(x: T[], ...ys: Iterable<T>): T[];
```

Reply:
Noted, thanks